### PR TITLE
Dat 3357 parse and print accurrences

### PR DESCRIPTION
**Pull request by royhadad**

### CODE REVIEW

File: pkg/printer/printer.go [hunk 1]
```diff
@@ -189,3 +194,11 @@ func (p *Printer) printPassedYamlValidation() {
 func (p *Printer) printSkippedPolicyCheck() {
 	p.printInColor("[?] Policy check didn’t run for this file\n", p.theme.Colors.White)
 }
+
+func (p *Printer) getStringOrNotAvailable(str string) string {
```
Comment (dimabru) [inline pkg/printer/printer.go:208]: Nice!!

File: pkg/printer/printer.go [hunk 2]
```diff
@@ -189,3 +194,11 @@ func (p *Printer) printPassedYamlValidation() {
...
+	if str == "" {
```
Comment (dimabru) [inline pkg/printer/printer.go:209]: Will this work if `str` is `undefined`?
I think we can address that as well

File: bl/evaluation/printer.go [hunk 3]
```diff
@@ -129,8 +129,9 @@ func parseToPrinterWarnings(results *EvaluationResults, invalidYamlFiles []*vali
 				rule := rules[ruleId]
 				details := printer.WarningInfo{
 					Caption:     rule.Name,
-					Occurrences: rule.Count,
+					Occurrences: rule.getCount(),
```
Comment (myishay) [inline bl/evaluation/printer.go]: Is the `getCount` available to you in this context? I guess not since it is not Capitalized

Comment (royhadad) [reply bl/evaluation/printer.go]: somehow it still worked, weird

File: pkg/printer/printer.go [hunk 4]
```diff
@@ -23,6 +24,7 @@ type WarningInfo struct {
 	Caption     string
 	Occurrences int
 	Suggestion  string
+	Matches     []*cliClient.Match
```
Comment (dimabru) [inline pkg/printer/printer.go]: I think `rule` object should not be aware of the concept of `matches`.
This is just the data type from the backend, but should not be part of the final object. I think saving a array of specific identifiers -> `metadataName` and `kind` is more accurate.

Matches includes rule-to-file mapping, while we are reversing the order of the print to file-to-rule mapping

**Review by dimabru — CHANGES_REQUESTED** [review]

File: pkg/printer/printer.go (hunk 2 above)
Comment (royhadad) [reply pkg/printer/printer.go:209]: in golang, string cannot be nil (only a pointer to a string can)

File: bl/evaluation/printer.go (hunk 3 above)
Comment (royhadad) [reply bl/evaluation/printer.go]: ohhh, I only used it in the "evaluation" package, so it still worked.
fixed it now.

**Review by dimabru — DISMISSED** [review]

**Pull request merged**